systemd: use Type=simple and no fork to avoid PIDFile race

mdomsch opened the pull request:
[Red Hat BZ#2056209](https://bugzilla.redhat.com/show_bug.cgi?id=2056209)

ghen2 commented:
This just stretches the race condition a little bit.

A better solution is to just drop the `PIDFile` line and let systemd manage the opendkim process directly (systemd recommends this way of operating).

For example on Arch Linux it's done this way: https://github.com/archlinux/svntogit-community/blob/packages/opendkim/trunk/opendkim.service

mdomsch commented:
@ghen2 fair play.  We can instead use type=simple and opendkim -f so it doesn't fork. Then we don't care about the PIDFIle at all either.

mdomsch committed:
systemd: use Type=simple and no forking to avoid PIDFile race

Red Hat Bugilla #2056209

mdomsch committed:
systemd: add Restart=on-abnormal

This is a long-running process that once started by systemd should
stay running forever. In the event of a fatal error and unclean exit,
we want systemd to restart the program. We use on-abnormal because
there are valid nonzero exit codes (EX_USAGE, EX_OSERR) which the
program can properly return indicating a configuration or command line
error, or inability to allocate memory early on. In these events,
restarting would be inappropriate, but on-failure would try anyhow.

Tachi107 commented:
Looks good!

I have submitted a similar patch to Debian, that also enables various hardening options: https://salsa.debian.org/debian/opendkim/-/merge_requests/3

I'll open a PR here too.

thegushi commented:
Superseded by PR #316, which combines this with the hardening options from #154 and the network-online.target fix from #141.
